python/Ganga/GPIDev/Lib/Registry/RegistrySlice.py

```diff
@@ -368,21 +373,17 @@ def __getitem__(self, x):
                 raise RegistryKeyError('object "%s" not found' % x)
             return addProxy(self.objects[ids[0]])
 
-        raise RegistryAccessError('Expected int or string (job name).')
+        if isinstance(x, slice):
+            start = x.start if x.start is not None else ''
+            stop = x.stop if x.stop is not None else ''
 
-    def __getslice__(self, i1, i2):
-        import sys
+            returnable = self.__class__("%s[%s:%s]" % (self.name, start, stop))
+            for id_ in self.ids()[x]:
+                returnable.objects[id_] = self.objects[id_]
 
-        if i2 == sys.maxsize:
-            endrange = ''
-        else:
-            endrange = str(i2)
+            return addProxy(returnable)
```

Comment:
No problem. I'm happy to play it safe here and look at removing these `addProxy` in the future.